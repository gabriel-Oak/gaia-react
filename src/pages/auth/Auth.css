.Auth form > .field_wrapper {
  display: flex;
  flex-direction: column;
}