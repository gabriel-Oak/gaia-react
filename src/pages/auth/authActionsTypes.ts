export interface AuthAction {
    type: string;
    value?: any;
}

export const AuthActionsTypes = {

}
