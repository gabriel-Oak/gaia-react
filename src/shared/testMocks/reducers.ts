import Action from "../../shared/interfaces/Action";
import { MainActionsTypes } from "../../pages/main/mainActionsTypes";

const MAIN_STATE = {
    title: 'Gaia',
    drawer: false,
    snackbar: {
        open: false,
    },
    user: {
        admin: true,
        email: 'test@email.cone',
        nome: 'Jorginho',
        user: 'string'
    },
}

export const mainReducerMock = (state = MAIN_STATE, action: Action) => {
    switch (action.type) {
        case MainActionsTypes.FETCH_USER:
            return { ...state, auth: true, user: action.value }
        default:
            return state;
    }
}
