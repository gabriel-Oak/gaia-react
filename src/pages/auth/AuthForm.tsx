import React from 'react'

import { Field, reduxForm } from 'redux-form';
import { required } from '../../shared/validators/validators';
import RenderTextField from '../../shared/components/renderTextField/renderTextField';

import { Button, CardActions, CardContent, CircularProgress } from '@material-ui/core';

import './Auth.css';

const AuthForm = (props: any) => {
  const { handleSubmit, onSubmit, loading } = props

  return (
    <form onSubmit={handleSubmit(onSubmit)}>

      <CardContent className="field_wrapper">
        <Field
          name="user"
          label="UsuÃ¡rio"
          variant="outlined"
          disabled={loading}
          component={RenderTextField}
          validate={[required]}
        />

        <Field
          name="senha"
          label="Senha"
          variant="outlined"
          disabled={loading}
          component={RenderTextField}
          validate={[required]}
        />
      </CardContent>

      <CardActions className="button-container">
        <Button
          color="primary"
          variant="contained"
          type="submit"
          disabled={loading}
        >
          {
            loading
              ? <CircularProgress size={24} />
              : 'Entrar'
          }
        </Button>
      </CardActions>
    </form>
  );
}

export default reduxForm({ form: 'auth' })(AuthForm);
