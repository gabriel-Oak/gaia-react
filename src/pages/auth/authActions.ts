import axios from 'axios';

import { Dispatch } from "redux";
import { AuthActionsTypes } from "./authActionsTypes";
import { api } from '../../enviroments/enviroments';
import resolveError from '../../shared/utils/resolveError';

export const logIn = (form: any) => async (dispatch: Dispatch) => {
  try {

    dispatch({ type: AuthActionsTypes.SENDED });

    const { data } = await axios.post(`${api.user}/${form.user}`, form);

    console.log(data);


  } catch (e) {
    dispatch({
      type: AuthActionsTypes.FIRE_SNACK,
      value: {
        open: true,
        type: 'error',
        message: resolveError(e).message,
        duration: 6000
      }
    })
  } finally {
    dispatch({ type: AuthActionsTypes.ENDED });
  }
}

export const closeSnack = () => ({
  type: AuthActionsTypes.CLOSE_SNACK
});