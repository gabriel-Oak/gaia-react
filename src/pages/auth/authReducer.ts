import { AuthAction, AuthActionsTypes } from './authActionsTypes';

export interface AuthState {
    loading: boolean;
    auth: boolean;
    snackbar: {
        open: boolean;
        message?: string;
        type?: 'success' | 'error' | 'info' | 'warning';
        duration?: number
    };
}

const INITIAL_STATE: AuthState = {
    loading: false,
    auth: false,
    snackbar: {
        open: false
    }
}

const authReducer = (state = INITIAL_STATE, action: AuthAction) => {
    switch (action.type) {
        case AuthActionsTypes.SENDED:
            return { ...state, loading: true }

        case AuthActionsTypes.ENDED:
            return { ...state, loading: false }

        case AuthActionsTypes.SUCCESS:
            return { ...state, auth: true }

        case AuthActionsTypes.FIRE_SNACK:
            return { ...state, snackbar: action.value }

        case AuthActionsTypes.CLOSE_SNACK:
            return { ...state, snackbar: { open: false } }

        default:
            return state;
    }
}

export default authReducer;
