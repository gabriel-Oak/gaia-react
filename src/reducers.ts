import { combineReducers } from 'redux';

import authReducer from './pages/auth/authReducer';
import mainReducer from './pages/main/mainReducer';

export interface ReducersPool {
  mainReducer: object;
  authReducer: object;
}

export default combineReducers({
  authReducer,
  mainReducer
});