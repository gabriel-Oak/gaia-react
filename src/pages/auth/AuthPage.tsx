import React, { PureComponent, ReactNode } from 'react'
import { ReducersPool } from '../../reducers';
import { AuthState } from './authReducer';
import { connect } from 'react-redux';

import { setTitle } from '../main/mainActions';

import { Card, CardHeader } from '@material-ui/core';
import AuthForm from './AuthForm';

interface Props extends AuthState {
  setTitle: Function;
}

class AuthPage extends PureComponent<Props> {
  constructor(props: Props) {
    super(props);

    this.props.setTitle('Login');
  }

  render(): ReactNode {
    return (
      <main className="Auth">
        <article className="Center-container s600">
          <Card>
            <CardHeader title="Login" />
            <AuthForm />
          </Card>
        </article>
      </main>
    )
  }
}

const mapStateToProps = (state: ReducersPool) => {
  const { authReducer } = state;
  return authReducer;
};

const mapDispatchToProps = {
  setTitle
};

export default connect(mapStateToProps, mapDispatchToProps)(AuthPage);
