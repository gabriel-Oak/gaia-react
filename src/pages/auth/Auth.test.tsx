import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';

import { createStore, applyMiddleware } from 'redux';
import reducers from '../../reducers';
import thunk from 'redux-thunk';

import AuthPage from './AuthPage';
import AuthForm from './AuthForm';
import { clearSession, setSession } from '../../shared/utils/auth';
import { configureShallow } from '../../shared/utils/testRender';
import { Provider } from 'react-redux';



describe('Auth', () => {
  let store: any;
  let page: any;

  const onSubmitMock = jest.fn();

  beforeEach(() => {
    store = createStore(
      reducers,
      applyMiddleware(thunk)
    );

    page = configureShallow(
      <AuthPage store={store} />
    );
  });

  afterEach(() => {
    clearSession();
  });

  it('should match snapshot', () => {
    expect(page).toMatchSnapshot();
  });

  it('should redirect when sessioned', () => {

    setSession('fakeToken');
    expect(
      configureShallow(
        <AuthPage store={store} />
      ).find('article').length
    ).toBe(0);

  });

  describe('Auth form', () => {

    it('should match snapshot', () => {

      const form = shallow(
        <AuthForm onSubmit={onSubmitMock} />
      ).dive().dive();

      expect(form).toMatchSnapshot();

    });

    it('should render the form without crashing', () => {

      const div = document.createElement('div');
      ReactDOM.render(
        <Provider store={store}>
          <AuthForm
            onSubmit={onSubmitMock}
          />
        </Provider>
        , div
      );
      ReactDOM.unmountComponentAtNode(div);

    });

    it('should render loading button', () => {

      const props = { loading: true };
      const div = document.createElement('div');

      ReactDOM.render(
        <Provider store={store}>
          <AuthForm
            onSubmit={onSubmitMock}
            {...props}
          />
        </Provider>
        , div
      );

      const loading = div.querySelector('.MuiCircularProgress-root');
      expect(loading).toBeTruthy();

    });
  });
});