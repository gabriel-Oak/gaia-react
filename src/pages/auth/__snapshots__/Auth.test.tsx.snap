// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Auth form should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="auth"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />
</ContextProvider>
`;

exports[`Auth should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(AuthPage) />
</ContextProvider>
`;
